activesupport/lib/active_support/fork_tracker.rb

```diff
@@ -35,10 +35,12 @@ def check!
       end
 
       def hook!
-        ::Object.prepend(CoreExtPrivate)
-        ::Kernel.prepend(CoreExtPrivate)
-        ::Kernel.singleton_class.prepend(CoreExt)
-        ::Process.singleton_class.prepend(CoreExt)
+        if Process.respond_to?(:fork)
```

Comment:
So the take away here is that if the process doesn't respond to fork then there's no need for the checker? Then how do we ensure we claim/reap the connections? cc @casperisfine

Reply:
Yup. When there's no `fork` available, there's no need to check and no need to claim/reap connections. E.g. on JRuby `fork` will always throw a "not implemented", so we're never in a situation that requires cleaning up connections

Reply:
Huh, TIL!